frontend/src/components/PlaceSearchSection.tsx

```diff
@@ -58,121 +60,136 @@ function PlaceSearchSection(props: PropType) {
             return {
               id: result.id,
               name: result.place_name,
-              description: result.place_name+'입니다',
+              description: result.place_name + "입니다",
               homepage: result.place_url,
               phone_number: result.phone,
               address: result.address_name,
               category: result.category_group_name,
               lon: result.x,
-              lat: result.y
-            }
+              lat: result.y,
+            };
           });
           setSearchResults(convertedPlaces);
         } else if (status === kakao.maps.services.Status.ZERO_RESULT) {
-          alert('검색 결과가 존재하지 않습니다.');
+          alert("검색 결과가 존재하지 않습니다.");
         } else if (status === kakao.maps.services.Status.ERROR) {
-          alert('검색 중 오류가 발생했습니다.');
+          alert("검색 중 오류가 발생했습니다.");
         }
-      }
+      };
       places.keywordSearch(searchTabQuery, placesSearchCB);
     }
   }, [isSearchRequested, searchTabQuery]);
 
   const isPlaceInCart = (id: number) => {
-    return cartPlaceList.some((p: PlaceType) => p.id === id);
-  }
+    return cartPlaceList.some((p: PlaceDayType) => p.place.id === id);
+  };
 
   const onClickCartButton = (place: PlaceType) => {
     if (isPlaceInCart(place.id)) {
-      setCartPlaceList(cartPlaceList.filter((p: PlaceType) => p.id !== place.id));
+      setCartPlaceList(
+        cartPlaceList.filter((p: PlaceDayType) => p.place.id !== place.id)
+      );
     } else {
-      setCartPlaceList([ ...cartPlaceList, place]);
+      setCartPlaceList([...cartPlaceList, { place: place, day: null }]);
```

Comment:
what is the type of `day`? I think it would be better if we manage the type strictly even when it has no value instead of just using `null`.

Reply:
I will describe the reason in detail at the last comment!